lib/knapsack_pro/store/server.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module KnapsackPro
+  module Store
+    class Server
+      extend Forwardable
+
+      def self.reset
+        stop
+        set_store_server_uri(nil)
+        @assigned_store_server_uri = nil
+        @client = nil
+      end
+
+      def self.start
+        return unless @server_pid.nil?
+
+        assign_available_store_server_uri
+
+        @server_pid = fork do
+          server_is_running = true
+
+          Signal.trap("TERM") {
+            server_is_running = false
+          }
+
+          DRb.start_service(store_server_uri, new)
+
+          while server_is_running
+            sleep 0.1
+          end
```

Comment:
That's an interesting article.

I've never heard about `ConditionVariable`.
https://ruby-doc.org/core-2.5.1/ConditionVariable.html
ChatGPT was good at explaining it to me.

I decided to use Queue because it's super simple: https://github.com/KnapsackPro/knapsack_pro-ruby/pull/252/commits/183f4ef25b62a18fa64bad548bcc2e6821dfbf80